File: scenarios/basic/mobile/flutter/SmallTouchTarget.dart
import 'package:flutter/material.dart';

class SmallTouchTarget extends StatefulWidget {
  const SmallTouchTarget({Key? key}) : super(key: key);

  @override
  State<SmallTouchTarget> createState() => _SmallTouchTargetState();
}

class _SmallTouchTargetState extends State<SmallTouchTarget> {
  int selectedTab = 0;
  bool isLiked = false;
  bool isBookmarked = false;
  int likeCount = 42;
  int shareCount = 8;

  final List<Map<String, dynamic>> posts = [
    {
      'id': 1,
      'username': 'john_doe',
      'userAvatar': 'https://via.placeholder.com/40x40',
      'postImage': 'https://via.placeholder.com/400x300',
      'caption': 'Beautiful sunset from my hike today! 🏔️ #nature #photography',
      'likes': 42,
      'comments': 8,
      'shares': 3,
      'timeAgo': '2h',
    },
    {
      'id': 2,
      'username': 'sarah_j',
      'userAvatar': 'https://via.placeholder.com/40x40',
      'postImage': 'https://via.placeholder.com/400x300',
      'caption': 'Just finished reading an amazing book! Highly recommend it to everyone.',
      'likes': 28,
      'comments': 12,
      'shares': 5,
      'timeAgo': '4h',
    },
    {
      'id': 3,
      'username': 'mike_chen',
      'userAvatar': 'https://via.placeholder.com/40x40',
      'postImage': 'https://via.placeholder.com/400x300',
      'caption': 'Working on a new project. Can\'t wait to share the results!',
      'likes': 15,
      'comments': 3,
      'shares': 1,
      'timeAgo': '6h',
    },
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.grey[50],
      appBar: AppBar(
        title: const Text('Social Feed'),
        backgroundColor: Colors.white,
        foregroundColor: Colors.black,
        elevation: 1,
        actions: [
          IconButton(
            icon: const Icon(Icons.search),
            onPressed: () {},
          ),
          IconButton(
            icon: const Icon(Icons.notifications),
            onPressed: () {},
          ),
        ],
      ),
      body: Column(
        children: [
          // Tab Navigation
          Container(
            color: Colors.white,
            child: Row(
              children: [
                _buildTab(0, 'Following'),
                _buildTab(1, 'Discover'),
                _buildTab(2, 'Trending'),
              ],
            ),
          ),
          
          // Posts Feed
          Expanded(
            child: ListView.builder(
              padding: const EdgeInsets.all(16),
              itemCount: posts.length,
              itemBuilder: (context, index) {
                return _buildPostCard(posts[index]);
              },
            ),
          ),
        ],
      ),
      bottomNavigationBar: BottomNavigationBar(
        currentIndex: 0,
        type: BottomNavigationBarType.fixed,
        items: const [
          BottomNavigationBarItem(
            icon: Icon(Icons.home),
            label: 'Home',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.search),
            label: 'Search',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.add_circle),
            label: 'Create',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.favorite),
            label: 'Activity',
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.person),
            label: 'Profile',
          ),
        ],
      ),
    );
  }

  Widget _buildTab(int index, String label) {
    final isSelected = selectedTab == index;
    return Expanded(
      child: GestureDetector(
        onTap: () {
          setState(() {
            selectedTab = index;
          });
        },
        child: Container(
          padding: const EdgeInsets.symmetric(vertical: 16),
          decoration: BoxDecoration(
            border: Border(
              bottom: BorderSide(
                color: isSelected ? Colors.blue : Colors.transparent,
                width: 2,
              ),
            ),
          ),
          child: Text(
            label,
            textAlign: TextAlign.center,
            style: TextStyle(
              fontSize: 16,
              fontWeight: isSelected ? FontWeight.bold : FontWeight.normal,
              color: isSelected ? Colors.blue : Colors.grey[600],
            ),
          ),
        ),
      ),
    );
  }

  Widget _buildPostCard(Map<String, dynamic> post) {
    return Container(
      margin: const EdgeInsets.only(bottom: 16),
      decoration: BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.circular(12),
        boxShadow: [
          BoxShadow(
            color: Colors.grey.withOpacity(0.1),
            spreadRadius: 1,
            blurRadius: 4,
            offset: const Offset(0, 2),
          ),
        ],
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          // Post Header
          Padding(
            padding: const EdgeInsets.all(16),
            child: Row(
              children: [
                CircleAvatar(
                  radius: 20,
                  backgroundImage: NetworkImage(post['userAvatar']),
                ),
                const SizedBox(width: 12),
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        post['username'],
                        style: const TextStyle(
                          fontSize: 16,
                          fontWeight: FontWeight.bold,
                          color: Colors.black87,
                        ),
                      ),
                      Text(
                        post['timeAgo'],
                        style: TextStyle(
                          fontSize: 12,
                          color: Colors.grey[600],
                        ),
                      ),
                    ],
                  ),
                ),
                // SMALL TOUCH TARGET - Only 20x20px on mobile interface
                GestureDetector(
                  onTap: () {
                    // More options
                  },
                  child: Container(
                    width: 20, // SMALL TOUCH TARGET
                    height: 20, // SMALL TOUCH TARGET
                    decoration: BoxDecoration(
                      color: Colors.grey[200],
                      shape: BoxShape.circle,
                    ),
                    child: const Icon(
                      Icons.more_horiz,
                      size: 12,
                      color: Colors.grey,
                    ),
                  ),
                ),
              ],
            ),
          ),
          
          // Post Image
          Container(
            width: double.infinity,
            height: 200,
            decoration: BoxDecoration(
              image: DecorationImage(
                image: NetworkImage(post['postImage']),
                fit: BoxFit.cover,
              ),
            ),
          ),
          
          // Post Actions
          Padding(
            padding: const EdgeInsets.all(16),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                // Action Buttons Row
                Row(
                  children: [
                    // Like Button - SMALL TOUCH TARGET
                    GestureDetector(
                      onTap: () {
                        setState(() {
                          isLiked = !isLiked;
                          if (isLiked) {
                            likeCount++;
                          } else {
                            likeCount--;
                          }
                        });
                      },
                      child: Container(
                        width: 20, // SMALL TOUCH TARGET
                        height: 20, // SMALL TOUCH TARGET
                        decoration: BoxDecoration(
                          color: Colors.transparent,
                        ),
                        child: Icon(
                          isLiked ? Icons.favorite : Icons.favorite_border,
                          size: 16,
                          color: isLiked ? Colors.red : Colors.grey[600],
                        ),
                      ),
                    ),
                    const SizedBox(width: 16),
                    
                    // Comment Button - SMALL TOUCH TARGET
                    GestureDetector(
                      onTap: () {
                        // Comment functionality
                      },
                      child: Container(
                        width: 20, // SMALL TOUCH TARGET
                        height: 20, // SMALL TOUCH TARGET
                        decoration: BoxDecoration(
                          color: Colors.transparent,
                        ),
                        child: Icon(
                          Icons.chat_bubble_outline,
                          size: 16,
                          color: Colors.grey[600],
                        ),
                      ),
                    ),
                    const SizedBox(width: 16),
                    
                    // Share Button - SMALL TOUCH TARGET
                    GestureDetector(
                      onTap: () {
                        setState(() {
                          shareCount++;
                        });
                      },
                      child: Container(
                        width: 20, // SMALL TOUCH TARGET
                        height: 20, // SMALL TOUCH TARGET
                        decoration: BoxDecoration(
                          color: Colors.transparent,
                        ),
                        child: Icon(
                          Icons.share,
                          size: 16,
                          color: Colors.grey[600],
                        ),
                      ),
                    ),
                    
                    const Spacer(),
                    
                    // Bookmark Button - SMALL TOUCH TARGET
                    GestureDetector(
                      onTap: () {
                        setState(() {
                          isBookmarked = !isBookmarked;
                        });
                      },
                      child: Container(
                        width: 20, // SMALL TOUCH TARGET
                        height: 20, // SMALL TOUCH TARGET
                        decoration: BoxDecoration(
                          color: Colors.transparent,
                        ),
                        child: Icon(
                          isBookmarked ? Icons.bookmark : Icons.bookmark_border,
                          size: 16,
                          color: isBookmarked ? Colors.blue : Colors.grey[600],
                        ),
                      ),
                    ),
                  ],
                ),
                
                const SizedBox(height: 8),
                
                // Like Count
                Text(
                  '$likeCount likes',
                  style: const TextStyle(
                    fontSize: 14,
                    fontWeight: FontWeight.bold,
                    color: Colors.black87,
                  ),
                ),
                
                const SizedBox(height: 8),
                
                // Caption
                RichText(
                  text: TextSpan(
                    children: [
                      TextSpan(
                        text: '${post['username']} ',
                        style: const TextStyle(
                          fontSize: 14,
                          fontWeight: FontWeight.bold,
                          color: Colors.black87,
                        ),
                      ),
                      TextSpan(
                        text: post['caption'],
                        style: const TextStyle(
                          fontSize: 14,
                          color: Colors.black87,
                        ),
                      ),
                    ],
                  ),
                ),
                
                const SizedBox(height: 8),
                
                // View Comments
                GestureDetector(
                  onTap: () {
                    // View comments
                  },
                  child: Text(
                    'View all ${post['comments']} comments',
                    style: TextStyle(
                      fontSize: 12,
                      color: Colors.grey[600],
                    ),
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

File: scenarios/basic/mobile/android-compose/SmallTouchTarget.kt
package com.example.accessibilityscenarios

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun SmallTouchTargetScreen() {
    val posts = listOf(
        SocialPost(
            id = 1,
            username = "john_doe",
            content = "Just finished an amazing hike! The views were incredible.",
            timestamp = "2 hours ago",
            likes = 24,
            comments = 8,
            shares = 3
        ),
        SocialPost(
            id = 2,
            username = "sarah_wilson",
            content = "Working on a new project. Can't wait to share it with everyone!",
            timestamp = "4 hours ago",
            likes = 156,
            comments = 23,
            shares = 12
        ),
        SocialPost(
            id = 3,
            username = "mike_chen",
            content = "Coffee break time! ☕",
            timestamp = "6 hours ago",
            likes = 89,
            comments = 15,
            shares = 5
        ),
        SocialPost(
            id = 4,
            username = "emma_taylor",
            content = "Beautiful sunset from my balcony tonight.",
            timestamp = "8 hours ago",
            likes = 203,
            comments = 31,
            shares = 18
        )
    )
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFFF5F5F5))
    ) {
        // Header
        Card(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            colors = CardDefaults.cardColors(containerColor = Color(0xFF1976D2))
        ) {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(16.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Social Feed",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White,
                    modifier = Modifier.weight(1f)
                )
                
                IconButton(
                    onClick = { }
                ) {
                    Icon(
                        imageVector = Icons.Default.Notifications,
                        contentDescription = "Notifications",
                        tint = Color.White
                    )
                }
            }
        }
        
        // Posts
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(horizontal = 16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            items(posts) { post ->
                PostCard(post = post)
            }
        }
    }
}

@Composable
fun PostCard(post: SocialPost) {
    Card(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp)),
        colors = CardDefaults.cardColors(containerColor = Color.White),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(
            modifier = Modifier.padding(16.dp)
        ) {
            // User Info
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(bottom = 12.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                // Profile Picture
                Box(
                    modifier = Modifier
                        .size(40.dp)
                        .background(Color(0xFFE0E0E0), RoundedCornerShape(20.dp)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Default.Person,
                        contentDescription = null,
                        tint = Color.Gray
                    )
                }
                
                Spacer(modifier = Modifier.width(12.dp))
                
                Column(
                    modifier = Modifier.weight(1f)
                ) {
                    Text(
                        text = post.username,
                        fontSize = 16.sp,
                        fontWeight = FontWeight.Medium
                    )
                    Text(
                        text = post.timestamp,
                        fontSize = 14.sp,
                        color = Color.Gray
                    )
                }
                
                // More options button - SMALL TOUCH TARGET
                IconButton(
                    onClick = { },
                    modifier = Modifier.size(20.dp) // TOO SMALL: Only 20dp touch target
                ) {
                    Icon(
                        imageVector = Icons.Default.MoreVert,
                        contentDescription = "More options",
                        modifier = Modifier.size(16.dp)
                    )
                }
            }
            
            // Post Content
            Text(
                text = post.content,
                fontSize = 16.sp,
                lineHeight = 24.sp,
                modifier = Modifier.padding(bottom = 16.dp)
            )
            
            // Action Buttons
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                // Like Button - SMALL TOUCH TARGET
                Row(
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    IconButton(
                        onClick = { },
                        modifier = Modifier.size(24.dp) // TOO SMALL: Only 24dp touch target
                    ) {
                        Icon(
                            imageVector = Icons.Default.ThumbUp,
                            contentDescription = "Like",
                            modifier = Modifier.size(18.dp),
                            tint = Color(0xFF1976D2)
                        )
                    }
                    Text(
                        text = "${post.likes}",
                        fontSize = 14.sp,
                        color = Color.Gray
                    )
                }
                
                // Comment Button - SMALL TOUCH TARGET
                Row(
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    IconButton(
                        onClick = { },
                        modifier = Modifier.size(24.dp) // TOO SMALL: Only 24dp touch target
                    ) {
                        Icon(
                            imageVector = Icons.Default.Comment,
                            contentDescription = "Comment",
                            modifier = Modifier.size(18.dp),
                            tint = Color.Gray
                        )
                    }
                    Text(
                        text = "${post.comments}",
                        fontSize = 14.sp,
                        color = Color.Gray
                    )
                }
                
                // Share Button - SMALL TOUCH TARGET
                Row(
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    IconButton(
                        onClick = { },
                        modifier = Modifier.size(24.dp) // TOO SMALL: Only 24dp touch target
                    ) {
                        Icon(
                            imageVector = Icons.Default.Share,
                            contentDescription = "Share",
                            modifier = Modifier.size(18.dp),
                            tint = Color.Gray
                        )
                    }
                    Text(
                        text = "${post.shares}",
                        fontSize = 14.sp,
                        color = Color.Gray
                    )
                }
            }
        }
    }
}

data class SocialPost(
    val id: Int,
    val username: String,
    val content: String,
    val timestamp: String,
    val likes: Int,
    val comments: Int,
    val shares: Int
)
